src/Analyzers/CSharp/Analyzers/UsePatternCombinators/CSharpUsePatternCombinatorsAnalyzer.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternCombinators
+{
+    using static BinaryOperatorKind;
+    using static AnalyzedPattern;
+
+    internal static class CSharpUsePatternCombinatorsAnalyzer
+    {
+        public static AnalyzedPattern? Analyze(IOperation operation)
+        {
+            var pattern = ParsePattern(operation);
+            return pattern?.Target.Syntax is ExpressionSyntax ? pattern : null;
+        }
+
+        private enum ConstantResult
+        {
+            /// <summary>
+            /// None of operands were constant.
+            /// </summary>
+            None,
+
+            /// <summary>
+            /// The left operand is the constant.
+            /// </summary>
+            Left,
+
+            /// <summary>
+            /// The right operand is the constant.
+            /// </summary>
+            Right,
+        }
+
+        private static AnalyzedPattern? ParsePattern(IOperation operation)
+        {
+            switch (operation)
```

Comment:
nit: consider switch expression :)

Reply:
I belive this is much cleaner since both patterns and values are kinda long.

Reply:
wfm